crates/bevy_ui/src/ui_node.rs

```diff
@@ -341,6 +353,361 @@ impl Style {
+    pub fn justify_content(mut self, val: JustifyContent) -> Self {
+        self.justify_content = val;
+        self
+    }
+
+    /// Set the `position.left` property
+    pub fn left(mut self, val: impl Into<Val>) -> Self {
+        self.position.left = val.into();
+        self
+    }
+    /// Set the `position.left` property
+    pub fn right(mut self, val: impl Into<Val>) -> Self {
+        self.position.right = val.into();
+        self
+    }
+    /// Set the `position.top` property
+    pub fn top(mut self, val: impl Into<Val>) -> Self {
+        self.position.top = val.into();
+        self
+    }
+    /// Set the `position.bottom` property
+    pub fn bottom(mut self, val: impl Into<Val>) -> Self {
+        self.position.bottom = val.into();
+        self
+    }
+    /// Set the `position.left` and `position.right` properties
+    pub fn position_horizontal(mut self, val: impl Into<Val>) -> Self {
+        let val = val.into();
+        self.position.right = val;
+        self.position.left = val;
+        self
+    }
+    /// Set the `position.top` and `position.bottom` properties
+    pub fn position_vertical(mut self, val: impl Into<Val>) -> Self {
+        let val = val.into();
+        self.position.top = val;
+        self.position.bottom = val;
+        self
+    }
+    /// Set the `position.top`,`position.bottom`,`position.left`, and `position.right` properties
+    pub fn position(mut self, val: impl Into<Val>) -> Self {
+        let val = val.into();
+        self.position.top = val;
+        self.position.bottom = val;
+        self.position.left = val;
+        self.position.right = val;
+        self
+    }
+
+    /// Set the `margin.left` property
+    pub fn margin_left(mut self, val: impl Into<Val>) -> Self {
+        self.margin.left = val.into();
+        self
+    }
+    /// Set the `margin.left` property
+    pub fn margin_right(mut self, val: impl Into<Val>) -> Self {
+        self.margin.right = val.into();
+        self
+    }
+    /// Set the `margin.top` property
+    pub fn margin_top(mut self, val: impl Into<Val>) -> Self {
+        self.margin.top = val.into();
+        self
+    }
+    /// Set the `margin.bottom` property
+    pub fn margin_bottom(mut self, val: impl Into<Val>) -> Self {
+        self.margin.bottom = val.into();
+        self
+    }
+    /// Set the `margin.left` and `margin.right` properties
+    pub fn margin_horizontal(mut self, val: impl Into<Val>) -> Self {
+        let val = val.into();
+        self.margin.right = val;
+        self.margin.left = val;
+        self
+    }
+    /// Set the `margin.top` and `margin.bottom` properties
+    pub fn margin_vertical(mut self, val: impl Into<Val>) -> Self {
+        let val = val.into();
+        self.margin.top = val;
+        self.margin.bottom = val;
+        self
+    }
+    /// Set the `margin.top`,`margin.bottom`,`margin.left`, and `margin.right` properties
+    pub fn margin(mut self, val: impl Into<Val>) -> Self {
```

Comment:
Just an idea, probably not good one (given the other methods), but does it make sense this to be `val: Into<UiRect>` and then allow:

- `42.` => `Px(42.0) Px(42.0) Px(42.0) Px(42.0)`
- `(42., ()) => `Px(42.0) Auto Auto Px(42.0)`
- `(10., 42.) => `Px(10.0) Px(42.0) Px(42.0) Px(10.0)`
- `((), (), Val::Percent(10.), ())` => `Auto Auto Val::Percent(10.0) Auto`

So this would give a similar API to CSS.

Reply:
In general I like the idea of supporting tuple parameters to this function. I'm not convinced by the idea of `()` meaning "auto" though. That seems non-obvious and potentially confusing to me. If `()` is supported, I think it should mean "don't mutate that value and leave it as whatever it already was".

I feel like an `auto()` function or `Auto` struct somewhere would be quite nice, but it does pollute the namespace. Potentially it could be imported like `use style_helpers::* as s` and then `s::Auto`. But at that point, perhaps it's no better than `Val::Auto`...

Reply:
Yeah, `()` could be quite confusing… 😅
Do you think using it as "don't mutate" placeholder would be that useful?

If we just had `f32 => Val::Px(f32)` and tuple parameters that already would be a great improvement. I don't think there is much gain with `s::*`… it's just 2 chars.

Reply:
> Do you think using it as "don't mutate" placeholder would be that useful?

Not really! I think `Rect::top(10.)` or similar would be much better. And in the few cases you have more complex requirements you can just write out the whole thing longhand.